fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,180 @@
+
+    Returns
+    -------
+    matplotlib.axes._axes.Axes
+    """
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+    # TODO: Check if metrics and error_bars/conf_intervals same length 
+
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+    
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    error_bars = [error_bars] if isinstance(error_bars, str) else error_bars
+
+    df = metric_frame.by_group
+
+    # only plot metrics that aren't tuples
+    if metrics is None:
+        metrics = []
+        for metric in list(df):
+            if not isinstance(df[metric][0], tuple) and metric != "_index":
+                metrics.append(metric)
+
+
+    if axs is None:
+        if subplot_shape is None:
+            subplot_shape = (1, len(metrics))
+        fig, axs = plt.subplots(*subplot_shape, squeeze=False, figsize=figsize)
+        axs = axs.flatten()
+
+    # plotting without error
+    df['_index'] = df.index
+    if error_bars is None and conf_intervals is None:
+        for metric, ax in zip(metrics, axs):
+            previous_xlabel = ax.get_xlabel()
+            ax = df.plot(x="_index", y=metric, kind=plot_type, ax=ax, colormap=colormap)
+            ax.set_xlabel(previous_xlabel)
+        del df['_index']
+        return axs
+
+    df_all_errors = pd.DataFrame([])
+    df_all_bounds = pd.DataFrame([])
+    # plotting with confidence intervals:
+    if conf_intervals is not None:
+        for metric, conf_interval in zip(metrics, conf_intervals):
+            df_all_errors[metric] = abs(df[metric] - df[conf_interval])
+
+            # assert bounds are valid
+            #assert (df_lower_error >= 0).all(), _NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR
+
+            if plot_error_labels:
+                df_all_bounds[metric] = df[conf_interval]
+    # plotting with relative errors
+    elif error_bars is not None:
+        for metric, error_bar in zip(metrics, error_bars):
+            df_all_errors[metric] = df[error_bar]
+
+            # assert bounds are valid
+            #assert (df_error >= 0).all(), _NEG_ERROR_OR_FLIPPED_BOUNDS_ERROR
+
+            if plot_error_labels:
+                df_error = pd.DataFrame([])
+                df_error[['lower', 'upper']] = pd.DataFrame(df[error_bar].tolist(), index=df.index)
+                df_error['error'] = list(zip(df[metric] - df_error['lower'], df[metric] + df_error['upper']))
+                df_all_bounds[metric] = df_error['error']
```

Comment:
This is more personal style, but you might want to mention that there's no `else` because that case is covered in the previous `if` statement. It's a point on which @romanlutz  and I may well disagree :-)

Reply:
I'm on the boat that it doesn't seem necessary, but I could include it if we want